crates/oxc_linter/src/rules/jest/prefer_mock_promise_shorthand.rs

```diff
@@ -0,0 +1,441 @@
+    ///   .mockReturnValueOnce(Promise.resolve(42))
+    ///   .mockImplementationOnce(() => Promise.resolve(42))
+    ///   .mockReturnValue(Promise.reject(new Error('too many calls!')));
+    ///
+    /// // valid
+    /// ```javascript
+    /// jest.fn().mockResolvedValue(123);
+    /// jest.spyOn(fs.promises, 'readFile').mockRejectedValue(new Error('oh noes!'));
+    ///
+    /// myFunction
+    ///   .mockResolvedValueOnce(42)
+    ///   .mockResolvedValueOnce(42)
+    ///   .mockRejectedValue(new Error('too many calls!'));
+    /// ```
+    ///
+    PreferMockPromiseShorthand,
+    correctness,
+);
+
+impl Rule for PreferMockPromiseShorthand {
+    fn run<'a>(&self, node: &oxc_semantic::AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Expression::MemberExpression(mem_expr) = &call_expr.callee else {
+            return;
+        };
+
+        if call_expr.arguments.is_empty() {
+            return;
+        }
+
+        let Some((property_span, property_name)) = mem_expr.static_property_info() else {
+            return;
+        };
+        let Some(Argument::Expression(expr)) = call_expr.arguments.first() else {
+            return;
+        };
+        let is_once = property_name.ends_with("Once");
+
+        if property_name.eq("mockReturnValue") || property_name.eq("mockReturnValueOnce") {
+            Self::report(is_once, property_span, None, expr, ctx);
+        } else if property_name.eq("mockImplementation")
+            || property_name.eq("mockImplementationOnce")
+        {
+            match expr {
+                Expression::ArrowFunctionExpression(arrow_func) => {
+                    if !arrow_func.params.is_empty() {
+                        return;
+                    }
+                    let Some(stmt) = arrow_func.body.statements.first() else {
+                        return;
+                    };
+                    match stmt {
+                        Statement::ReturnStatement(return_stmt) => {
+                            let Some(arg_expr) = &return_stmt.argument else {
+                                return;
+                            };
+                            Self::report(
+                                is_once,
+                                property_span,
+                                Some(arrow_func.span),
+                                arg_expr,
+                                ctx,
+                            );
+                        }
+                        Statement::ExpressionStatement(expr_stmt) => {
+                            Self::report(
+                                is_once,
+                                property_span,
+                                Some(arrow_func.span),
+                                &expr_stmt.expression,
+                                ctx,
+                            );
+                        }
+                        _ => (),
```

Comment:
If you want to get the return expression of arrow expression, try [`get_expression`](https://github.com/oxc-project/oxc/blob/75226f3b4ad99290aebf38d2fb96483a72bcdfa3/crates/oxc_ast/src/ast/js.rs#L1949) if [`expression`](https://github.com/oxc-project/oxc/blob/75226f3b4ad99290aebf38d2fb96483a72bcdfa3/crates/oxc_ast/src/ast/js.rs#L1938-L1939) is `true`.

[playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICsgICAgICAgICxG0qZRraXD3k8CFzRo%2FGwCMKuF%2FdswcD7%2FCfE2%2FB%2BlkU7WtaOX3zVtIA%3D)

Reply:
Yes, but I still need to check another type(`const () => {}`) of arrow function, if I didn't misunderstand this [#comment](https://github.com/oxc-project/oxc/blob/75226f3b4ad99290aebf38d2fb96483a72bcdfa3/crates/oxc_ast/src/ast/js.rs#L1938). So I think I have checked it, just write more code to support another.

Reply:
Apologized for my naive. I didn't realize it might occur performance bug. 😣